Pipfile

```diff
@@ -3,13 +3,16 @@ url = "https://pypi.python.org/simple"
 verify_ssl = true
 name = "pypi"
 
+
 [packages]
 numpy = "*"
 pygithub = "*"
 matplotlib = "*"
 pandas = "*"
 thoth-storages = "*"
 semver = "*"
+idna_ssl = "*"
+typing_extensions = "*"
```

Comment:
Are these required direct dependencies?

Reply:
This was a problem with `pytest`, but I still don't know why.
```
======================================================================
ERROR: bot_knowledge (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: bot_knowledge
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/loader.py", line 153, in loadTestsFromName
    module = __import__(module_name)
  File "/workspace/repo/srcopsmetrics/bot_knowledge.py", line 27, in <module>
    from srcopsmetrics.entities import Entity, NOT_FOR_INSPECTION
  File "/workspace/repo/srcopsmetrics/entities/__init__.py", line 19, in <module>
    from .interface import Entity
  File "/workspace/repo/srcopsmetrics/entities/interface.py", line 32, in <module>
    from srcopsmetrics.entities.tools.storage import KnowledgeStorage
  File "/workspace/repo/srcopsmetrics/entities/tools/storage.py", line 26, in <module>
    from thoth.storages.ceph import CephStore
  File "/opt/app-root/lib/python3.6/site-packages/thoth/storages/__init__.py", line 30, in <module>
    from .graph import GraphDatabase
  File "/opt/app-root/lib/python3.6/site-packages/thoth/storages/graph/__init__.py", line 21, in <module>
    from .postgres import GraphDatabase
  File "/opt/app-root/lib/python3.6/site-packages/thoth/storages/graph/postgres.py", line 53, in <module>
    from thoth.python import PackageVersion
  File "/opt/app-root/lib/python3.6/site-packages/thoth/python/__init__.py", line 29, in <module>
    from .aiosource import AIOSource, AsyncIterablePackages, AsyncIterableVersions, AsyncIterableArtifacts
  File "/opt/app-root/lib/python3.6/site-packages/thoth/python/aiosource.py", line 29, in <module>
    import aiohttp
  File "/opt/app-root/lib/python3.6/site-packages/aiohttp/__init__.py", line 6, in <module>
    from .client import (
  File "/opt/app-root/lib/python3.6/site-packages/aiohttp/client.py", line 35, in <module>
    from . import hdrs, http, payload
  File "/opt/app-root/lib/python3.6/site-packages/aiohttp/http.py", line 7, in <module>
    from .http_parser import (
  File "/opt/app-root/lib/python3.6/site-packages/aiohttp/http_parser.py", line 15, in <module>
    from .helpers import NO_EXTENSIONS, BaseTimerContext
  File "/opt/app-root/lib/python3.6/site-packages/aiohttp/helpers.py", line 62, in <module>
    import idna_ssl
ModuleNotFoundError: No module named 'idna_ssl'
```

The typing_extensions was a similar error, catched by local `pytest` check:

```
======================================================================
ERROR: iterator (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: iterator
Traceback (most recent call last):
  File "/usr/lib64/python3.8/unittest/loader.py", line 154, in loadTestsFromName
    module = __import__(module_name)
  File "/home/xtuchyna/git/forks/SrcOpsMetrics/srcopsmetrics/iterator.py", line 30, in <module>
    from srcopsmetrics.entities import Entity
  File "/home/xtuchyna/git/forks/SrcOpsMetrics/srcopsmetrics/entities/__init__.py", line 19, in <module>
    from .interface import Entity
  File "/home/xtuchyna/git/forks/SrcOpsMetrics/srcopsmetrics/entities/interface.py", line 32, in <module>
    from srcopsmetrics.entities.tools.storage import KnowledgeStorage
  File "/home/xtuchyna/git/forks/SrcOpsMetrics/srcopsmetrics/entities/tools/storage.py", line 26, in <module>
    from thoth.storages.ceph import CephStore
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib/python3.8/site-packages/thoth/storages/__init__.py", line 30, in <module>
    from .graph import GraphDatabase
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib/python3.8/site-packages/thoth/storages/graph/__init__.py", line 21, in <module>
    from .postgres import GraphDatabase
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib/python3.8/site-packages/thoth/storages/graph/postgres.py", line 53, in <module>
    from thoth.python import PackageVersion
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib/python3.8/site-packages/thoth/python/__init__.py", line 29, in <module>
    from .aiosource import AIOSource, AsyncIterablePackages, AsyncIterableVersions, AsyncIterableArtifacts
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib/python3.8/site-packages/thoth/python/aiosource.py", line 26, in <module>
    import aiohttp
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib64/python3.8/site-packages/aiohttp/__init__.py", line 6, in <module>
    from .client import (
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib64/python3.8/site-packages/aiohttp/client.py", line 35, in <module>
    from . import hdrs, http, payload
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib64/python3.8/site-packages/aiohttp/http.py", line 7, in <module>
    from .http_parser import (
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib64/python3.8/site-packages/aiohttp/http_parser.py", line 15, in <module>
    from .helpers import NO_EXTENSIONS, BaseTimerContext
  File "/home/xtuchyna/.local/share/virtualenvs/SrcOpsMetrics-Vwmk3Tpf/lib64/python3.8/site-packages/aiohttp/helpers.py", line 48, in <module>
    from typing_extensions import Protocol
ModuleNotFoundError: No module named 'typing_extensions'
```
After adding them, these issues were fixed.

Reply:
it seems that this happend after `thoth-storages` update, look here #299

Reply:
Did you try deleting and recreating Pipfile.lock but without adding those two to direct dependencies?

Reply:
@pacospace problem solved :beers: